agent/consul/state/config_entry.go

```diff
@@ -371,6 +373,100 @@ var serviceGraphKinds = []string{
 	structs.ServiceResolver,
 }
 
+// discoveryChainTargets will return a list of services listed as a target for the input's discovery chain
+func (s *Store) discoveryChainTargets(ws memdb.WatchSet, dc, service string, entMeta *structs.EnterpriseMeta) (uint64, []structs.ServiceName, error) {
+	source := structs.NewServiceName(service, entMeta)
+	req := discoverychain.CompileRequest{
+		ServiceName:         source.Name,
+		EvaluateInNamespace: source.NamespaceOrDefault(),
+
+		// TODO(freddy) : Should these be anything other than the known DC?
+		EvaluateInDatacenter: dc,
```

Comment:
For ordinary proxycfg work, this is set to whatever the `datacenter` field on an upstream is defined as.

* `EvaluateInDatacenter`:  which datacenter should I do this discovery request against?
* `UseInDatacenter`: which datacenter are you dialing from?

Reply:
This is used in determining things like "should this use a mesh gateway?" and selectively omitting options from the failover groups if you're already in one.

Reply:
OK, so I think it should be the local DC for both here.

Reply:
And if it turns out you need to adjust it, it's perfectly fine to fix it up later.